packages/cli/src/ceramic-daemon.ts

```diff
@@ -295,8 +295,13 @@ export class CeramicDaemon {
     }
   }
 
+  // Mark Ceramic unhealthy if associated IPFS is not 'online'
   healthcheck(req: Request, res: Response): void {
-    res.status(200).send('Alive!')
+    if (this.ceramic.ipfs.isOnline()) {
```

Comment:
please make sure to test this with both an in-process and out-of-process (http-client) ipfs node.

Reply:
Will do 👍🏼

Reply:
Needed to add some additional code after all for out-of-process IPFS node. Worked correctly if Ceramic came up before IPFS but then it kept returning 200 even after I killed IPFS.

With the latest change, if IPFS dies later on, the health-check will fail and mark Ceramic as unhealthy. I'm wondering if we should do that or not...

Doing so would mean the Ceramic node being marked unhealthy if IPFS crashes, for example, which might cause Ceramic to also be recreated but maybe that's not a bad thing?

Reply:
hmm that's a good point. I don't really know what's best practice around whether or not a service should report itself as healthy if a service it in turn depends on is unhealthy.  Would be a good question for our future Observability Engineer.  In lieu of that person, let's wait for Val to come back and weigh in.

I'm also going to add @oed and @ukstv to the review

Reply:
> Doing so would mean the Ceramic node being marked unhealthy if IPFS crashes, for example, which might cause Ceramic to also be recreated but maybe that's not a bad thing?

Well that means that streams would be marked as not synced and it would cause a lot more pubsub traffic.

>  Would be a good question for our future Observability Engineer.

What is it that that person could answer but we can't now?

Reply:
Is there a way to handle IPFS being offline more gracefully without marking Ceramic as unhealthy? I do need to test what would actually happen with the current code - I suspect Ceramic would be restarted if it becomes unhealthy but will double-check.

Assuming that Ceramic also restarts when unhealthy, perhaps it's ok to just postpone "healthy" during init to ensure the right startup seq and then not care if IPFS goes offline after that?

Reply:
> if IPFS dies later on, the health-check will fail and mark Ceramic as unhealthy
> 
> Doing so would mean the Ceramic node being marked unhealthy if IPFS crashes, for example, which might cause Ceramic to also be recreated but maybe that's not a bad thing?

This is the current behavior. If IPFS dies after Ceramic starts then Ceramic dies. I don't know what part of the code causes this but that's what happens today.

I think this behavior makes sense. As @ukstv said below, they are a package deal.

Reply:
> Is there a way to handle IPFS being offline more gracefully without marking Ceramic as unhealthy?
Why? Can Ceramic operate in a healthy state without IPFS?

Reply:
> Why? Can Ceramic operate in a healthy state without IPFS?

It can't but restarting the Ceramic node does have other undesirable side-effects, e.g. the one @oed mentioned:

> Well that means that streams would be marked as not synced and it would cause a lot more pubsub traffic.

Avoiding this would be a good reason to not restart Ceramic when IPFS restarts, and instead to error for API calls (since Ceramic can't work when IPFS is down) and then wait till IPFS comes back up to allow APIs to work again.

Ceramic and IPFS are definitely a package deal but Ceramic doesn't need to be restarted to ensure that it's only used when IPFS is also online.

I don't want to bikeshed too much 😅 so I'd be fine with merging the code as it is (i.e. Ceramic becomes unhealthy and restarts if IPFS does) and then coming back to it later if y'all are ok with that. Just wanted to point out that there _is_ a tradeoff to consider here. @v-stickykeys @ukstv

Reply:
@smrz2001 thanks for explanation and yeah that makes sense. I don't understand what "streams would be marked as not synced means"

But it seems like there is a way to gracefully shut down--i.e. maybe Ceramic needs to do some clean up before it shuts down if IPFS can not be reached. Will leave it to you to decide if that should be in this PR or come later.